src/app/examples/todos/todos.reducer.ts

```diff
@@ -14,7 +14,9 @@ export enum TodosActionTypes {
 export class ActionTodosAdd implements Action {
   readonly type = TodosActionTypes.ADD;
 
-  constructor(readonly payload: { name: string }) {}
+  constructor(public payload: { id?: string; name: string }) {
```

Comment:
If someone would pass an `id` this would be overridden.
To quickly fix this we can do:
```ts
this.payload.id =  this.payload.id || uuid();
```
But personally I like this approach more:
```ts
readonly payload: { id: string, name: string };

constructor({ id = uuid(), name = '' }: {id?: string, name: string}) {
  this.payload = { id, name };
}
```

Or if you don't want to "allow" passing the `id`  property:

```ts
readonly payload: { id: string, name: string };

constructor({ name }: {name: string}) {
  this.payload = { id: uuid(), name };
}
```